### DISCUSSION TITLE
Real Time Quality Control (RTQC) vs. Delayed Mode Quality Control (DMQC)

### DISCUSSION THREAD

**Discussion by soerenthomsen** [GliderToolsCommunity/GliderTools#97] [General]
Hi GliderTools Community, 

I was told that not for everyone the difference between **Real Time Quality Control (RTQC)** and **Delayed Mode Quality Control (DMQC)** is clear. Thus I though that a separate thread would be helpful to explain the differences + people can ask questions or say for what they have already tools to share / make compatible with GliderTools etc.

**RTQC:** Gliders which are out in the ocean and directly send their data to data centres in "real time", the data has to be quality checked within short time. A lot of these procedures thus have to be automatic but still require operators to check whether everything is fine etc. Gui @castelao is an expert for this and maybe you can expand, mention examples or docs and the packages you have developed? 

**DMQC:** These are corrections done typically after the deployment which take more time but thus result in the highest data quality. This data is typically used by scientists for their publications etc. i.e. for oxygen #74 you aim to calibrate your optodes/foils before and after the deployment at different temperatures i.e. in the cold/warm lab and/or by attaching the optodes to the CTD + Winkler etc and make multiple calibration stops. These procedures give you calibration points which you need to get high quality oxygen data. For other parameters there are other approaches. For example in the optics package of GliderTools which is quite advanced there is a [`quenching correction`](https://glidertools.readthedocs.io/en/latest/optics.html) applied following [Thomalla et al. 2017](https://aslopubs.onlinelibrary.wiley.com/doi/abs/10.1002/lom3.10234). 

Please let me know whether this is clear. I made an overview figure here #107, which might help to develop a joint vision of a future package structure, i.e. where it makes sense to make new developments and/or merge with existing tools. 

@dhruvbalwada @jbusecke @lukegre @tjryankeogh @isgiddy

**Comment by castelao** [comment]
@soerenthomsen , it was a good explanation! It is probably worth adding that RTQC is not only about doing it fast but most of the time we don't have access to all the information available for the DMQC. Once we recover the glider we can access all records on full resolution, thus we can do a better job. For RTQC we do the best we can with the information that we have now, while DMQC could be months after the measurement until it is available. RTQC can also be used to support the pilots with automatic alerts.

I think that the Argo Program is a fantastic model on this subject.
